plugins/vpp/ifplugin/vppcalls/if_vppcalls.go

```diff
@@ -108,6 +108,18 @@ type InterfaceSpanDetails struct {
 	IsL2          uint8
 }
 
+// GreTunnelDetails is something
+type GreTunnelDetails struct {
+	SwIfIndex  uint32
+	Instance   uint32
+	IsIPv6     uint8
+	TunnelType uint8
+	SrcAddress string
```

Comment:
Using type `net.IP` here, will allow you to get rid of `IsIPv6`.

Reply:
not sure if this will help. As I understand, you are saying about checking `IsIPv6` in dump function:
```go
if greDetails.IsIPv6 == 1 { ...
```

But in `greDetails` addresses comes as `[]byte` and, for example, ip `10.10.10.10` will be presented as `[10 10 10 10 0 0 0 0 0 0 0 0 0 0 0 0]`, on the other hand ip `2019::8:23` will be presented as `[32 25 0 0 0 0 0 0 0 0 0 0 0 8 0 35]`.

If I did not check `IsIPv6` and not slice this slice if it's IPv4, then `String()` method on such `net.IP` in a case with `10.10.10.10` will return `a0a:a0a::` and this is definitely not what I've expected to see.

One thing is actually redundant is calling `To4()` and `To16()`, because slice was already trimmed to proper size.

Reply:
What I meant was to use the `net.IP` values parsed in dump: 
https://github.com/ligato/vpp-agent/blob/cd0dc983bbb0a6ccdb613f7ef57274abe5da2bfd/plugins/vpp/ifplugin/vppcalls/vpp1904/gre_vppcalls.go#L108-L109

And store as `net.IP` here:
https://github.com/ligato/vpp-agent/blob/cd0dc983bbb0a6ccdb613f7ef57274abe5da2bfd/plugins/vpp/ifplugin/vppcalls/vpp1904/gre_vppcalls.go#L120-L121

And the caller who calls dump can simply use the addresses directly without parsing them from string again. You can easily check if `net.IP` is IPv4 or IPv6:

```go
	ip4 := net.ParseIP("10.0.1.1")
	fmt.Println(ip4)
	fmt.Printf("is IPv6: %v\n\n", ip4.To4() == nil) // prints false
	
	ip6 := net.ParseIP("fe80::200:5aee:feaa:20a2")
	fmt.Println(ip6)
	fmt.Printf("is IPv6: %v\n", ip6.To4() == nil) // prints true
```
> https://play.golang.org/p/8WQKPMVmHJz

Reply:
oh I see now. Yes, it would be better to keep IP addresses as `net.IP`